so_vector/track.py

```diff
@@ -98,18 +106,34 @@ def params(self):
             if "filter" in self._params:
                 result["body"]["query"]["script_score"]["query"] = self._params["filter"]
         else:
-            result["body"] = {
-                "knn": {
-                    "field": "titleVector",
-                    "query_vector": query_vec,
-                    "k": self._params.get("k", 10),
-                    "num_candidates": self._params.get("num_candidates", 50),
+            if self._in_esql_mode:
+                # Construct options JSON.
+                options_param = '{"num_candidates":' + str(self._num_candidates)
```

Comment:
This will change to "min_candidates" in https://github.com/elastic/elasticsearch/pull/132944. `k` will also be removed, and instead we will need to specify a LIMIT to the query.

It's not a blocker for this PR, but just a heads up that we will need to change this when the above gets merged.